---
title: "comparing light response with different genotype volcano plots"
output: html_document
---

```{r message=FALSE, warning=FALSE, tidy=TRUE}
library(ggplot2)
library(dplyr)

prepare_data_for_graph <- function(dds.results.table){
  dds.results.table$highlight <- "not significant"
  dds.results.table$highlight[dds.results.table$padj < 0.1] <- "significant"
  dds.results.table$highlight <- factor(dds.results.table$highlight, 
																			  levels = c("not significant", "significant"))
  return(dds.results.table)
}

```

```{r message=FALSE, warning=FALSE, tidy=TRUE}

vocanlo_plot <- function(dds.results.table, title){
  gpr88 <- dds.results.table[dds.results.table$mgi_symbol %in% "Gpr88",]

  # Volcano plot
  g <- ggplot(dds.results.table, aes(x=log2FoldChange, y= -log10(pvalue)))
  g <- g + geom_jitter(aes(color = highlight)) + scale_color_manual(values = c("grey", "black"))
  g <- g + ylim(0, 10) + xlim(-8, 8)
  g <- g + xlab("log2 (Fold change)") + ggtitle(title)
  g <- g +  theme(text = element_text(size=16),
								  panel.background = element_rect(fill = NA),
								  axis.line = element_line(size = 0.8, colour = "black"),
								  axis.ticks = element_line(size = 1),
								  legend.position = c(0.89, 0.95),
								  legend.key = element_rect(size = 0.5, fill = "white", colour = "black"),
								  legend.title = element_blank(),
  								plot.title = element_text(hjust = 0.5)) 
  
  g <- g + geom_point(aes(x = gpr88$log2FoldChange , y = -log10(gpr88$pvalue)), color = "red")
  g <- g + annotate(geom = "label", x = gpr88$log2FoldChange + 0.8, y = -log10(gpr88$pvalue)+0.3, label = "Gpr88", color="red")

return(g)
}
```

```{r message=FALSE, warning=FALSE, tidy=TRUE}
# load the dataframe with the gene info and remark added
setwd("~/Desktop/brainRNASeq/R_analysis/R_output/results_tables")
SCN <- read.csv("SCN/SCN_effectAcrossGenotype.csv", header = TRUE, stringsAsFactors = FALSE)
VTA <- read.csv("VTA/VTA_effectAcrossGenotype.csv", header = TRUE, stringsAsFactors = FALSE)
LHb <- read.csv("LHb/LHb_effectAcrossGenotype.csv", header = TRUE, stringsAsFactors = FALSE)
NAc <- read.csv("NAc/NAc_effectAcrossGenotype.csv", header = TRUE, stringsAsFactors = FALSE)

```

```{r fig.height=7, fig.width=7, message=FALSE, warning=FALSE}
SCN.updated <- prepare_data_for_graph(SCN)
vocanlo_plot(SCN.updated, "SCN")
```

```{r fig.height=7, fig.width=7, message=FALSE, warning=FALSE}
VTA.updated <- prepare_data_for_graph(VTA)
vocanlo_plot(VTA.updated, "VTA")
```

```{r fig.height=7, fig.width=7, message=FALSE, warning=FALSE}
LHb.updated <- prepare_data_for_graph(LHb)
vocanlo_plot(LHb.updated, "LHb")
```

```{r fig.height=7, fig.width=7, message=FALSE, warning=FALSE}
NAc.updated <- prepare_data_for_graph(NAc)
vocanlo_plot(NAc.updated, "NAc")
```