---
title: "NAc vocanlo plots"
output: html_document
---

```{r message=FALSE, warning=FALSE, tidy=TRUE}
library(ggplot2)
library(dplyr)

prepare_data_for_graph <- function(dds.results.table){
  dds.results.table <- dds.results.table %>% filter(log2FoldChange < 10 & log2FoldChange > -10) %>%
	filter(-log10(pvalue) < 10)

  dds.results.table$highlight <- "not significant"
  dds.results.table$highlight[dds.results.table$padj < 0.1] <- "significant"
  dds.results.table$highlight[dds.results.table$mgi_symbol %in% "Gpr88"] <- "Gpr88"
  dds.results.table$highlight[dds.results.table$mgi_symbol %in% "Drd3"] <- "Drd3"
  dds.results.table$highlight <- factor(dds.results.table$highlight, 
																			  levels = c("not significant", "significant", "Drd3", "Gpr88"))
  return(dds.results.table)
}

```

```{r fig.height=5, fig.width=7, message=FALSE, warning=FALSE}

vocanlo_plot <- function(dds.results.table){
	drd3 <- dds.results.table[dds.results.table$mgi_symbol %in% "Drd3",]
  gpr88 <- dds.results.table[dds.results.table$mgi_symbol %in% "Gpr88",]

  # Volcano plot
  g <- ggplot(dds.results.table, aes(x=log2FoldChange, y= -log10(pvalue)))
  g <- g + geom_jitter(aes(color = highlight)) + scale_color_manual(values = c("grey", "black", "red", "green"))

  g <- g + xlab("log2 (Fold change)")

  g <- g + geom_point(aes(x = drd3$log2FoldChange , y = -log10(drd3$pvalue)), color = "red")
  g <- g + geom_point(aes(x = gpr88$log2FoldChange , y = -log10(gpr88$pvalue)), color = "green")


  g <- g +  theme(text = element_text(size=16),
								  axis.title.y = element_text(size=16),
								  panel.background = element_rect(fill = NA),
								  axis.line = element_line(size = 0.8, colour = "black"),
								  axis.ticks = element_line(size = 1),
								  legend.position = c(0.89, 0.90),
								  legend.key = element_rect(size = 0.5, fill = "white", colour = "black"),
								  legend.title = element_blank()) 
return(g)
}
```

```{r}
# load the dataframe with the gene info and remark added
setwd("~/Desktop/brainRNASeq/R_analysis/R_output/results_tables/NAc")
NAc.1 <- read.csv("NAc_WT_lightvsControl.csv", header = TRUE, stringsAsFactors = FALSE)
NAc.2 <- read.csv("NAc_Per1_lightvsControl.csv", header = TRUE, stringsAsFactors = FALSE)
NAc.3 <- read.csv("NAc_WTvsPer1_control.csv", header = TRUE, stringsAsFactors = FALSE)
NAc.4 <- read.csv("NAc_WTvsPer1_light.csv", header = TRUE, stringsAsFactors = FALSE)
NAc.5 <- read.csv("NAc_effectAcrossGenotype.csv", header = TRUE, stringsAsFactors = FALSE)
```

## WT Light vs control
```{r fig.height=7, fig.width=7}
NAc.1.updated <- prepare_data_for_graph(NAc.1)
vocanlo_plot(NAc.1.updated)
```

## Per1 Light vs control
```{r fig.height=7, fig.width=7}
NAc.2.updated <- prepare_data_for_graph(NAc.2)
vocanlo_plot(NAc.2.updated)
```

## WT vs Per1 (control)
```{r fig.height=7, fig.width=7}
NAc.3.updated <- prepare_data_for_graph(NAc.3)
vocanlo_plot(NAc.3.updated)
```

## WT vs Per1 (light)
```{r fig.height=7, fig.width=7}
NAc.4.updated <- prepare_data_for_graph(NAc.4)
vocanlo_plot(NAc.4.updated)
```

## the effect of genotype on light treatment
```{r fig.height=7, fig.width=7}
NAc.5.updated <- prepare_data_for_graph(NAc.5)
vocanlo_plot(NAc.5.updated)
```

```{r}
sessionInfo()
```

