---
title: "Venn diagrams"
output: html_document
---
```{r message=FALSE, warning=FALSE, tidy=TRUE}
library(VennDiagram)
setwd("~/Desktop/brainRNASeq/R_analysis/R_output/results_tables/")

significant_genes <- function(dds.results.table){
  dds.results.table$remark <- "not significant"
  dds.results.table$remark [dds.results.table$padj < 0.1] <- "significant"
  dds.results.table$remark  <- factor(dds.results.table$remark, 
																			  levels = c("not significant", "significant"))
  return(dds.results.table)
}

VTA_effectAcrossGenotype <- read.csv("VTA/VTA_effectAcrossGenotype.csv", header = TRUE, stringsAsFactors = FALSE)
VTA_WT <- read.csv("VTA/VTA_WT_lightvsControl.csv", header = TRUE, stringsAsFactors = FALSE)
VTA_Per1 <- read.csv("VTA/VTA_Per1_lightvsControl.csv", header = TRUE, stringsAsFactors = FALSE)

VTA_effectAcrossGenotype <- significant_genes(VTA_effectAcrossGenotype)
VTA_WT <- significant_genes(VTA_WT)
VTA_Per1 <- significant_genes(VTA_Per1)

NAc_effectAcrossGenotype <- read.csv("NAc/NAc_effectAcrossGenotype.csv", header = TRUE, stringsAsFactors = FALSE)
NAc_WT <- read.csv("NAc/NAc_WT_lightvsControl.csv", header = TRUE, stringsAsFactors = FALSE)
NAc_Per1 <- read.csv("NAc/NAc_Per1_lightvsControl.csv", header = TRUE, stringsAsFactors = FALSE)

NAc_effectAcrossGenotype <- significant_genes(NAc_effectAcrossGenotype)
NAc_WT <- significant_genes(NAc_WT)
NAc_Per1 <- significant_genes(NAc_Per1)
```
## VTA
```{r message=FALSE, warning=FALSE, tidy=TRUE}
VTA.effect.genotype <- VTA_effectAcrossGenotype$ensembl_gene_id[VTA_effectAcrossGenotype$remark %in% "significant"]
VTA.WT.genes <- VTA_WT$ensembl_gene_id[VTA_WT$remark %in% "significant"]
VTA.Per1.genes <- VTA_Per1$ensembl_gene_id[VTA_Per1$remark %in% "significant"]

venn.plot1 <- venn.diagram(
  x = list(VTA.effect.genotype, VTA.WT.genes, VTA.Per1.genes),
  category.names = c("interaction" , "WT " , "Per1"),
  filename = NULL,
  output=TRUE
)
grid.draw(venn.plot1)
```

## NAc
```{r message=FALSE, warning=FALSE, tidy=TRUE}
NAc.effect.genotype <- NAc_effectAcrossGenotype$ensembl_gene_id[NAc_effectAcrossGenotype$remark %in% "significant"]
NAc.WT.genes <- NAc_WT$ensembl_gene_id[NAc_WT$remark %in% "significant"]
NAc.Per1.genes <- NAc_Per1$ensembl_gene_id[NAc_Per1$remark %in% "significant"]

venn.plot2 <- venn.diagram(
  x = list(NAc.effect.genotype, NAc.WT.genes, NAc.Per1.genes),
  category.names = c("interaction" , "WT " , "Per1"),
  filename = NULL
)
grid.draw(venn.plot2)
```

## WT VTA vs NAc
```{r message=FALSE, warning=FALSE, tidy=TRUE}
venn.plot3 <- venn.diagram(
  x = list(VTA.WT.genes, NAc.WT.genes),
  category.names = c("VTA" , "NAc"),
  filename = NULL
)
grid.draw(venn.plot3)
```

## Per1 VTA vs NAc
```{r message=FALSE, warning=FALSE, tidy=TRUE}
venn.plot4 <- venn.diagram(
  x = list(VTA.Per1.genes, NAc.Per1.genes),
  category.names = c("VTA" , "NAc"),
  filename = NULL
)
grid.draw(venn.plot4)
```

## interaction VTA vs NAc
```{r message=FALSE, warning=FALSE, tidy=TRUE}
venn.plot5 <- venn.diagram(
  x = list(VTA.effect.genotype, NAc.effect.genotype),
  category.names = c("VTA" , "NAc"),
  filename = NULL
)
grid.draw(venn.plot5)
```