---
title: "SCN vocanlo plots"
output: html_document
---

```{r message=FALSE, warning=FALSE, tidy=TRUE}
library(ggplot2)
library(dplyr)

prepare_data_for_graph <- function(dds.results.table){
  dds.results.table <- dds.results.table %>% filter(log2FoldChange < 10 & log2FoldChange > -10) %>%
	filter(-log10(pvalue) < 10)

  dds.results.table$highlight <- "not significant"
  dds.results.table$highlight[dds.results.table$padj < 0.1] <- "significant"
  dds.results.table$highlight <- factor(dds.results.table$highlight, 
																			  levels = c("not significant", "significant"))
  return(dds.results.table)
}

```

```{r fig.height=5, fig.width=7, message=FALSE, warning=FALSE}

vocanlo_plot <- function(dds.results.table){
  gpr88 <- dds.results.table[dds.results.table$mgi_symbol %in% "Gpr88",]

  # Volcano plot
  g <- ggplot(dds.results.table, aes(x=log2FoldChange, y= -log10(pvalue)))
  g <- g + geom_jitter(aes(color = highlight)) + scale_color_manual(values = c("grey", "black"))
  g <- g + ylim(0, 10) + xlim(-8, 8)
  g <- g + xlab("log2 (Fold change)")
  g <- g + geom_point(aes(x = gpr88$log2FoldChange , y = -log10(gpr88$pvalue)), color = "green")
  g <- g +  theme(text = element_text(size=16),
								  axis.title.y = element_text(size=16),
								  panel.background = element_rect(fill = NA),
								  axis.line = element_line(size = 0.8, colour = "black"),
								  axis.ticks = element_line(size = 1),
								  legend.position = c(0.89, 0.90),
								  legend.key = element_rect(size = 0.5, fill = "white", colour = "black"),
								  legend.title = element_blank()) 
  g <- g + annotate(geom = "label", x = gpr88$log2FoldChange + 0.8, y = -log10(gpr88$pvalue)+0.3, label = "Gpr88", color="green")

return(g)
}
```

```{r}
# load the dataframe with the gene info and remark added
setwd("~/Desktop/brainRNASeq/R_analysis/R_output/results_tables/SCN")
SCN.1 <- read.csv("SCN_WT_lightvsControl.csv", header = TRUE, stringsAsFactors = FALSE)
SCN.2 <- read.csv("SCN_Per1_lightvsControl.csv", header = TRUE, stringsAsFactors = FALSE)
SCN.3 <- read.csv("SCN_WTvsPer1_control.csv", header = TRUE, stringsAsFactors = FALSE)
SCN.4 <- read.csv("SCN_WTvsPer1_light.csv", header = TRUE, stringsAsFactors = FALSE)
SCN.5 <- read.csv("SCN_effectAcrossGenotype.csv", header = TRUE, stringsAsFactors = FALSE)
```

```{r fig.height=7, fig.width=7, message=FALSE, warning=FALSE}
SCN.1.updated <- prepare_data_for_graph(SCN.1)
vocanlo_plot(SCN.1.updated)

```

```{r fig.height=7, fig.width=7}
SCN.2.updated <- prepare_data_for_graph(SCN.2)
vocanlo_plot(SCN.2.updated)
```

```{r fig.height=7, fig.width=7}
SCN.3.updated <- prepare_data_for_graph(SCN.3)
vocanlo_plot(SCN.3.updated)
```