---
title: "Vocanlo plot draft (SCN, showing Grp88, Drd3)"
output: html_document
---
```{r message=FALSE, warning=FALSE, tidy=TRUE}
library(ggplot2)
library(dplyr)

# load the dataframe with the gene info and remark added
setwd("~/Desktop/brainRNASeq/R_analysis/R_output/results_tables/SCN")
dds.results.table <- read.csv("SCN_effectAcrossGenotype.csv", header = TRUE,
															 stringsAsFactors = FALSE)

dds.results.table$remark <- "not significant"
dds.results.table$remark[dds.results.table$padj < 0.1] <- "significant"
dds.results.table$remark <- as.factor(dds.results.table$remark)
```

```{r}
gpr88 <- dds.results.table[dds.results.table$mgi_symbol %in% "Gpr88",]
drd3 <- dds.results.table[dds.results.table$mgi_symbol %in% "Drd3",]
```

```{r}
# Volcano plot
g <- ggplot(dds.results.table, aes(x=log2FoldChange, y= -log10(pvalue)))
g <- g + geom_jitter(aes(color = remark), alpha = I(1/3)) + scale_color_manual(values = c("grey", "black"))
g <- g + theme_classic()
g <- g + geom_point(aes(x = gpr88$log2FoldChange , y = -log10(gpr88$pvalue)), color = "red")
#g <- g + geom_label(label="Gpr88", 
  #                  x=5,
   #                 y=-log10(gpr88$pvalue),
    #                color = "red")
#g <- g + geom_point(aes(x = drd3$log2FoldChange , y = -log10(drd3$pvalue)), color = "blue")
#g <- g + geom_label(label="Drd3", 
 #                   x=5,
  #                  y=1,
   #                 color = "blue")
print(g)
# 
```