---
title: "DESeq2 analysis visualization - LHb"
output: html_document
---

```{r message=FALSE, warning=FALSE, tidy=TRUE}
library(DESeq2)
library(tximport)
library(readr)
library(rhdf5)

# read the csv file containing the metadata 
meta.data <- read_csv("~/Desktop/brainRNASeq/R_analysis/input_doc/meta_data_LHb.csv")

# read in the files with the list of transcipt ID corresponding to gene ID
tx2gene <- read_csv("~/Desktop/brainRNASeq/R_analysis/input_doc/tx2gene.csv")

# define the directory with the kallisto outputs
kallisto.output.directory <- "~/Desktop/brainRNASeq/pseudoalignment"

# get the paths of the kallisto outputs (abundance.h5 files) for all the samples 
# and test if the files exists
abundance.h5.files <- file.path(kallisto.output.directory, "output", 
																meta.data$file_name, "abundance.h5")
names(abundance.h5.files) <- meta.data$sample
file.exists(abundance.h5.files)

# use tximport to combine the abundance.h5 files, and conventing the transcript 
# ID into gene ID
abundance.all.samples <- tximport(abundance.h5.files, type = "kallisto", 
																	tx2gene = tx2gene)

# define the experimental design 
experimental.design <- ~ genotype + condition + genotype:condition

# creating the DESeqDataSet object(dds) from Tximport
dds <- DESeqDataSetFromTximport(abundance.all.samples, meta.data, 
																design = experimental.design)

# relevel to make sure that the reference is wildtype 'WT'
dds$genotype <- relevel(dds$genotype, "WT")

# filtering out the genes that have too low expression 
dds <- dds[rowSums(counts(dds)) > 25, ]

# creating the DESeqDataSet object again
dds <- DESeq(dds)

# get the names for comparsion between different experimental conditions
resultsNames(dds)
```

```{r message=FALSE, tidy=TRUE, fig.height=8, fig.width=7}
plotDispEsts(dds, ylim = c(1e-6, 1e1))
```

```{r message=FALSE, tidy=TRUE, fig.height=8, fig.width=7}
par(mar=c(8,5,2,2))
boxplot(log10(assays(dds)[["cooks"]]), range=0, las=2)
```

PCA and the overview of the difference between samples
```{r message=FALSE, tidy=TRUE, fig.height=8, fig.width=7}

library(ggplot2)
library(RColorBrewer)
library(ClassDiscovery) 

rlog.transformed.dds  <- rlog(dds)

pca.data <- plotPCA(rlog.transformed.dds , 
										intgroup=c("genotype", "condition"), 
										returnData=TRUE)

percentVar <- round(100 * attr(pca.data, "percentVar"))

ggplot(pca.data, aes(PC1, PC2, color=genotype,shape=condition)) +
                 geom_point(size=3) +
                 xlab(paste0("PC1: ",percentVar[1],"% variance")) +
                 ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
                 coord_fixed()
```


```{r fig.height=6, fig.width=6}
# calculate the sample distance
sample.distance <- dist(t(assay(rlog.transformed.dds )))

# convert the sample distance into a matrix
distance.matrix <- as.matrix(sample.distance)

# define the rownames
rownames(distance.matrix) <- paste(rlog.transformed.dds$genotype, 
																	 rlog.transformed.dds$condition, sep="-")

# define the colour
colours = colorRampPalette(rev(brewer.pal(9, "Blues")))(255)

# plot the heatmap
aspectHeatmap(distance.matrix, labCol=NULL,
							                 col = colours, 
							                 hExp=2, 
							                 wExp=1.4, 
							                 margins=c(6,3))
```


```{r}
sessionInfo()
```