---
title: "Differential gene expression visualization - SCN WTvsPer1_control"
output: html_document
---

Comparing the WT and Per1 without light treatment(control)

```{r message=FALSE, warning=FALSE, tidy=TRUE}
library(DESeq2)
library(ggplot2)

# load the DESeqResult object
setwd("~/Desktop/brainRNASeq/R_analysis/R_output/DESeq2Objects")
dds.results <- readRDS("SCN/SCN_WTvsPer1_control.RData")
```

```{r}
plotMA(dds.results, ylim = c(-2, 2))
```

```{r}
metadata(dds.results)$filterThreshold
```

```{r}
plot(metadata(dds.results)$filterNumRej, xlab="quantile of baseMean", 
		 ylab="number of rejections")
```

```{r}
hist(dds.results$pvalue, breaks = 20, col ='grey', 
		 main = "WT vs Per1 mutant (control)", xlab = "pvalue", ylab = "Frequency")
```

```{r warning=FALSE}
# load the dataframe with the gene info and remark added
setwd("~/Desktop/brainRNASeq/R_analysis/R_output/results_tables/SCN")
dds.results.table <- read.csv("SCN_WTvsPer1_control.csv", header = TRUE,
															 stringsAsFactors = FALSE)

dds.results.table$remark <- as.factor(dds.results.table$remark)

summary(dds.results.table$remark)
```

```{r}
# Volcano plot
ggplot(dds.results.table, aes(x=log2FoldChange, y= -log10(pvalue))) + 
	                         geom_jitter(aes(color = remark)) + 
	                         ggtitle("WT vs Per1 mutant (control) Volcano plot")
```

```{r}
library(ClassDiscovery) 
library(RColorBrewer)

setwd("~/Desktop/brainRNASeq/R_analysis/R_output/results_tables/SCN")
rlog.transformated <- read.csv("SCN_rlog_transformated_data.csv")

# subset the list of ensembl_gene_id of genes that are not "not significant"
significant.genes <- dds.results.table[dds.results.table$log2FoldChange > 1,] 
significant.genes.name <- significant.genes$ensembl_gene_id

# subset the rlog transformed data with the significant genes list
rlog.significant.genes <- rlog.transformated[ rlog.transformated$X %in% significant.genes.name,]
rownames(rlog.significant.genes) <- rlog.significant.genes$X
rlog.significant.genes <- rlog.significant.genes[,-1]
rlog.significant.genes <- as.matrix(rlog.significant.genes)

aspectHeatmap(rlog.significant.genes, labCol=NULL, Colv = NA,
							                 hExp=2, 
							                 wExp=1.4, 
							                 margins=c(6,3))

```

```{r}
sessionInfo()
```